compiler/quilt/nodes.py

```diff
@@ -127,12 +122,11 @@ def _data(self, asa=None):
         Merges all child dataframes. Only works for dataframes stored on disk - not in memory.
         """
         if asa is not None and not callable(asa):
```

Comment:
Same as in `DataNode`, you can drop this check and the assert below, and let Python do its thing.